packages/dingtalk/default.nix

```diff
@@ -0,0 +1,146 @@
+{ stdenv
+, fetchurl
+, autoPatchelfHook
+, makeWrapper
+, lib
+  # DingTalk dependencies
+, alsa-lib
+, at-spi2-atk
+, at-spi2-core
+, cairo
+, cups
+, dbus
+, e2fsprogs
+, gdk-pixbuf
+, glib
+, gnutls
+, graphite2
+, gtk2
+, libdrm
+, libgcrypt
+, libGLU
+, libpulseaudio
+, libthai
+, libxkbcommon
+, mesa_drivers
+, nspr
+, nss
+, openldap
+, rtmpdump
+, udev
+, util-linux
+, xorg
+, ...
+} @ args:
+
+################################################################################
+# Mostly based on dingtalk-bin package from AUR:
+# https://aur.archlinux.org/packages/dingtalk-bin
+################################################################################
+
+let
+  version = "1.3.0.20214";
+
+  libraries = [
+    alsa-lib
+    at-spi2-atk
+    at-spi2-core
+    cairo
+    cups
+    dbus
+    e2fsprogs
+    gdk-pixbuf
+    glib
+    gnutls
+    graphite2
+    gtk2
+    libdrm
+    libgcrypt
+    libGLU
+    libpulseaudio
+    libthai
+    libxkbcommon
+    mesa_drivers
+    nspr
+    nss
+    openldap
+    rtmpdump
+    udev
+    util-linux
+    xorg.libICE
+    xorg.libSM
+    xorg.libX11
+    xorg.libxcb
+    xorg.libXcomposite
+    xorg.libXcursor
+    xorg.libXdamage
+    xorg.libXext
+    xorg.libXfixes
+    xorg.libXi
+    xorg.libXinerama
+    xorg.libXmu
+    xorg.libXrandr
+    xorg.libXrender
+    xorg.libXScrnSaver
+    xorg.libXt
+    xorg.libXtst
+  ];
+in
+stdenv.mkDerivation rec {
+  pname = "dingtalk-resource";
```

Comment:
Why not use "dingtalk" as the pname? Users can also use `nix run` to use this derivation directly if the pname matches the binary name.